### DISCUSSION TITLE
Filter object in an array and getting a value

### DISCUSSION THREAD

**Discussion by Disura-Randunu** [schibsted/jslt#227] [Q&A] [answered]
Hello, JSLT Team.
I want to know that if it's possible to filter an array of objects based on a key : value pair and get something out of that filtered object. Is this possible? If so, can you give an example?

Payload I'm trying to process:
```
"measures": [
      {
        "metric": "sqale_rating",
        "value": "1.0",
        "bestValue": true
      },
      {
        "metric": "reliability_rating",
        "value": "3.0",
        "bestValue": false
      }
]
```
I want to filter and get the object item based on the 'metric' value. It would be great if you can give an example using this.

**Comment by larsga** [comment, marked as answer]
```
[for (.measures)
  .
 if (.metric == "sqale_rating")]
```

will give you

```
[
     {
        "metric": "sqale_rating",
        "value": "1.0",
        "bestValue": true
      }
]
```

Does that answer your question?

**Reply by Disura-Randunu** [reply]
@larsga, Thank you very much for the answer. I'm really sorry for the late reply. I got it working when I tried as you suggested. Is it also possible to extract the object itself from the array so that the final output will be like below:
 ```
{
        "metric": "sqale_rating",
        "value": "1.0",
        "bestValue": true
}
```

**Reply by Disura-Randunu** [reply]
I found the answer to the above. Using array indexing I was able to extract.
```
let obj = [for (.measures) . if (.metric == "sqale_rating")]

{
    "metric_obj": $obj[0]
}
```
If there is a direct way to extra it without declaring a variable, it would be great. Is this possible?

**Reply by larsga** [reply]
It's not possible yet, but issue #96 is about fixing it. It's something we want to support, we just don't yet.
